pkg/syncer/cnsoperator/controller/cnsnodevmattachment/cnsnodevmattachment_controller.go

```diff
@@ -405,38 +406,45 @@ func (r *ReconcileCnsNodeVMAttachment) Reconcile(ctx context.Context,
 						"CnsNodeVmAttachment request with name: %q on namespace: %q",
 						nodeUUID, request.Name, request.Namespace)
 					recordEvent(ctx, r, instance, v1.EventTypeWarning, msg)
-					return reconcile.Result{RequeueAfter: timeout}, nil
+					return reconcile.Result{RequeueAfter: timeout}, "NotComputed", nil
 				}
 				// Now that VM on VC is not found, check VirtualMachine CRD instance exists.
 				// This check is needed in scenarios where VC inventory is stale due
 				// to upgrade or back-up and restore.
-				isPresent, err := isVmCrPresent(ctx, r.vmOperatorClient, nodeUUID,
+				isPresent, vmInstance, err := isVmCrPresent(ctx, r.vmOperatorClient, nodeUUID,
 					request.Namespace)
 				if err != nil {
 					msg = fmt.Sprintf("failed to verify is VM CR is present with UUID: %s "+
 						"in namespace: %s", nodeUUID, request.Namespace)
 					recordEvent(ctx, r, instance, v1.EventTypeWarning, msg)
-					return reconcile.Result{RequeueAfter: timeout}, nil
+					return reconcile.Result{RequeueAfter: timeout}, "NotComputed", nil
 				}
 				if !isPresent {
 					msg = fmt.Sprintf("VM CR is not present with UUID: %s in namespace: %s. "+
-						"Removing finalizer on CnsNodeVMAttachment: %s instance.",
+						"Removing finalizer on CnsNodeVMAttachment: %s instance and "+
+						"returning success since the deletion timestamp is set.",
```

Comment:
Can you remove the text `since the deletion timestamp is set` from the log? That will be misleading as we have not yet checked the deletion timestamp on VM CR.

Reply:
Removed